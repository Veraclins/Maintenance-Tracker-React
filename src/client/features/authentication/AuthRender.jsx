import React from 'react';
import PropTypes from 'prop-types';

import Forms from '../../shared/components/Form';

const AuthRender = ({ self }) => {
  const {
    props: properties,
    handleChange,
    handleSubmit,
    state,
    action,
  } = self;
  const { errors } = properties;
  const isLogin = action === 'Login';
  const formLinks = {
    label: isLogin ? "Don't have an account?" : 'Already Registered?',
    link: isLogin ? 'signup' : 'login',
    text: isLogin ? 'Create One' : 'Login',
  };
  return (
    <Forms
      handleInputChange={handleChange}
      inputs={state}
      handleSubmit={handleSubmit}
      errors={errors}
      formLinks={formLinks}
      formTitle={isLogin ? 'Log into your Account' : 'Create an Account'}
    />
  );
};

AuthRender.propTypes = {
  self: PropTypes.shape({}).isRequired,
};

export default AuthRender;
