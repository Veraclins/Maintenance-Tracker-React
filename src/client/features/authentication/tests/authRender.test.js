import React from 'react';
import { shallow } from 'enzyme';

import AuthRender from '../AuthRender';
import FormComponent from '../../../shared/components/Form';

const self = {
  props: {},
};

describe('Tests the login component', () => {
  it('should render without errors', () => {
    const wrapper = shallow(<AuthRender
      self={self}
    />);
    expect(wrapper.find(FormComponent)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
