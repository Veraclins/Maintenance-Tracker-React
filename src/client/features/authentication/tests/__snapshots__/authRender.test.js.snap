// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the login component should render without errors 1`] = `
<Form
  formLinks={
    Object {
      "label": "Already Registered?",
      "link": "login",
      "text": "Login",
    }
  }
  formTitle="Create an Account"
  handleSelectChange={[Function]}
  handleTextAreaChange={[Function]}
  selects={Object {}}
  textAreas={Object {}}
/>
`;
